modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/display/context/CommercePriceEntryDisplayContext.java

```diff
@@ -180,7 +180,7 @@ public List<FDSActionDropdownItem> getPriceEntriesFDSActionDropdownItems()
 	public String getPriceEntryApiURL() throws PortalException {
 		return "/o/headless-commerce-admin-pricing/v2.0/price-lists/" +
 			getCommercePriceListId() +
-				"/price-entries?nestedFields=product,sku";
+				"/price-entries?nestedFields=measurementUnit,product,sku";
```

Comment:
I don't think you need this, please sync with https://github.com/liferay-commerce/liferay-portal/pull/4185

Reply:
The rest looks good to me

Reply:
> I don't think you need this, please sync with #4185

Hey @andrea-ale-sbarra I took a look at https://github.com/liferay-commerce/liferay-portal/pull/4185, from what I can see only unitOfMeasureKey is being passed through the API. We need to be able to access the name so we can display it on the UOM field. Im not sure if there is an other way of getting that name without including the measurementUnit in the  API response.

Unless we change from using frontend-data-set:headless-display to frontend-data-set:classic-display [here](https://github.com/liferay-commerce/liferay-portal/blob/3067ef6501f1793efb29bef6d5a0b176dc74ed20/modules/apps/commerce/commerce-pricing-web/src/main/resources/META-INF/resources/commerce_price_lists/price_entries.jsp#L95C5-L95C40) then we could create a FDSDataProvider for price list and promotions.

Let me know what you think, thanks!

Reply:
Now I see your point, you're trying to print the name of the UOM.. I believe the best would be to add the name to the PriceEntry and TierPrice entity both in DB and in the headless API, but I'm not 100% sure the requirement says to print the name and even if it does I'm not sure we can correct this now, it would be easier and quicker to manage this during the stabilisation. Anyway I wouldn't add a new endpoint just to print a name..